Update document for handle SageMaker throttling (#7331)
opensearch-project/documentation-website: _ml-commons-plugin/remote-models/blueprints.md

| `post_process_function` | String   | Optional. A built-in or custom Painless script used to post-process the model output data. OpenSearch provides the following built-in post-process functions that you can call directly:<br> - `connector.pre_process.cohere.embedding` for [Cohere text embedding models](https://docs.cohere.com/reference/embed)<br> - `connector.pre_process.openai.embedding` for [OpenAI text embedding models](https://platform.openai.com/docs/api-reference/embeddings) <br> - `connector.post_process.default.embedding`, which you can use to post-process documents in the model response so that they are in the format that neural search expects (OpenSearch 2.11 or later). For more information, see [Built-in functions](#built-in-pre--and-post-processing-functions). |

The `client_config` parameter supports the following options.

| Field  | Data type | Description |
|:---|:---|:---|

> | `max_connection` | Integer   | The maximum number of concurrent connections that the client can establish with the server. Some remote services like SageMaker constrain the maximum number of concurrent connections and throw a throttling exception if the number of concurrent connections exceeds the threshold. The max concurrent connections from OpenSearch is `max_connection`*`node_number_for_connector`. To mitigate this issue, try to decrease the value of this parameter, and modify the retry settings in `client_config`. Default is `30`. |

Reviewer: ```suggestion
| `max_connection` | Integer   | The maximum number of concurrent connections that the client can establish to the server. Some remote services, like SageMaker, constrain the maximum number of concurrent connections and throw a throttling exception if the number of concurrent connections exceeds the threshold. The maximum number of concurrent OpenSearch connections is `max_connection`*`node_number_for_connector`. To mitigate this issue, try to decrease the value of this parameter and modify the retry settings in `client_config`. Default is `30`. |
```

> | `connection_timeout` | Integer | The maximum amount of time (in seconds) that the client will wait while trying to establish a connection to the server. A timeout prevents the client from waiting indefinitely and allows the client to recover when encountering unreachable network endpoints. |

Reviewer: ```suggestion
| `connection_timeout` | Integer | The maximum amount of time (in seconds) that the client will wait while trying to establish a connection to the server. A timeout prevents the client from waiting indefinitely and allows the client to recover when it encounters unreachable network endpoints. |
```

> | `read_timeout` | Integer | The maximum amount of time (in seconds) that the client will wait for a response from the server after sending a request. This is useful when the server is slow to respond or encounters issues while processing a request. |

Reviewer: ```suggestion
| `read_timeout` | Integer | The maximum amount of time (in seconds) that the client will wait for a response from the server after sending a request. This is useful when the server is slow to respond or encounters an issue while processing a request. |
```

> | `retry_backoff_policy`  | String   | The backoff policy for retries to the remote connector. This is useful when there is spike in traffic causing throttling exceptions. Supported policies are `constant`, `exponential_equal_jitter`, `exponential_full_jitter`. Default is `constant`. |

Reviewer: ```suggestion
| `retry_backoff_policy`  | String   | The backoff policy for retries to the remote connector. This is useful when there is spike in traffic causing throttling exceptions. Supported policies are `constant`, `exponential_equal_jitter`, and `exponential_full_jitter`. Default is `constant`. |
```

> | `max_retry_times`  | Integer   | The maximum retry times for a single remote inference request. This is useful when there is a spike in traffic causing throttling exceptions. When set to `0`, the retry is disabled. When set as `-1`, OpenSearch does not limit the retry by the number of `retry_times`. Setting this setting to positive integers means sets the maximum number of retry attempts. Default is `0`.       |

Reviewer: ```suggestion
| `max_retry_times`  | Integer   | The maximum number of times that a single remote inference request will be retried. This is useful when there is a spike in traffic causing throttling exceptions. When set to `0`, retrying is disabled. When set to `-1`, OpenSearch does not limit the number of `retry_times`. Setting this to a positive integer specifies the maximum number of retry attempts. Default is `0`.       |
```

> | `retry_backoff_millis` | Integer   | The base backoff time in milliseconds for retry policy. The suspend time during two retries is determined by this parameter and `retry_backoff_policy`.  Default is `200`. |

Reviewer: ```suggestion
| `retry_backoff_millis` | Integer   | The base backoff time, in milliseconds, for the retry policy. The suspend time during two retries is determined by this parameter and the `retry_backoff_policy`. Default is `200`. |
```

> | `retry_backoff_millis` | Integer   | The base backoff time in milliseconds for retry policy. The suspend time during two retries is determined by this parameter and `retry_backoff_policy`.  Default is `200`. |

Reviewer: "suspend" => "suspension"?

> | `retry_timeout_seconds` | Integer   | The timeout value in seconds to end the retry. If the retry can not succeed in the specified time, the connector will stop retrying and throw exceptions. Default is `30`. |

Reviewer: ```suggestion
| `retry_timeout_seconds` | Integer   | The timeout value, in seconds, for the retry. If the retry can not succeed within the specified amount of time, the connector will stop retrying and throw an exception. Default is `30`. |
```
